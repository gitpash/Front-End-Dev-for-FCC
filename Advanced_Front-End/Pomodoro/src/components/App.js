import React from "react";
import ReactDOM from "react-dom";
import PropTypes from "prop-types";
class App extends React.Component {
  constructor(props) {
    super(props);
    this.handleClick = this.handleClick.bind(this);
    this.handleActive = this.handleActive.bind(this);
    this.state = {
      displayCount: 25,
      isRunning: "active"
    };
  }
  handleClick() {
    const { displayCount, isRunning } = this.state;
    displayCount > 0
      ? this.setState({
          displayCount: displayCount - 1
        })
      : this.setState({ displayCount: 0 });
      setInterval(this.handleClick, 1000)
  }
  
  handleActive() {
    this.state.isRunning === "active"
      ? this.setState({ isRunning: "inActive" })
      : this.setState({ isRunning: "active" });
  }
  render() {
    const { displayCount, isRunning } = this.state;
    return (
      <div className="app-container">
        <div className="box-timer">
          <div
            onClick={
              isRunning === "active"
                ? () => this.handleClick()
                : () => this.handleActive()
            }
            className="circle"
          >
            <p>{displayCount}</p>
          </div>
        </div>
      </div>
    );
  }
}

export default App;
