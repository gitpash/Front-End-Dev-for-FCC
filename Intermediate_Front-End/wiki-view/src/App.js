import React, { Component } from "react";
import api from "./api";
import PropTypes from "prop-types";

const Search = ({ value, onSubmit, onChange }) => (
  <form onSubmit={onSubmit}>
    <input type="text" value={value} onChange={onChange} autoFocus />
    <button type="submit">
      go!
    </button>
  </form>
);

const ArticlesGrid = props => (
  <ul className="wrapper-list">
    {props.title.map((title, index) => (
      <a href={props.url[index]} target="_blank" key={props.title[index]}>
        <ul className="articles-block">
          <h3>{title}</h3>
          <li className="article">
            <p>{props.description[index]}</p>
          </li>
        </ul>
      </a>
    ))}
  </ul>
);

ArticlesGrid.propTypes = {
  url: PropTypes.array.isRequired,
  title: PropTypes.array.isRequired,
  description: PropTypes.array.isRequired
};

class App extends Component {
  constructor(props) {
    super(props);
    // связываем любые изменения списка с ре-рендером компонента
    this.state = {
      result: null,
      searchTerm: "",
      title: null,
      description: null,
      url: null
    };
    this.onSearchChange = this.onSearchChange.bind(this);
    this.onSearchSubmit = this.onSearchSubmit.bind(this);
  }

  onSearchChange(event) {
    this.setState({ searchTerm: event.target.value });
  }
  onSearchSubmit(event) {
    const { searchTerm } = this.state;
    this.setState({ searchKey: searchTerm });

    this.searchByRequest(searchTerm);
    event.preventDefault();
  }

  searchByRequest(searchTerm) {
    api.fetchArticles(this.state.searchTerm).then(
      function(articles) {
        this.setState(function() {
          return { title: articles[1] };
        });
        this.setState(function() {
          return {
            description: articles[2]
          };
        });
        this.setState(function() {
          return {
            url: articles[3]
          };
        });
      }.bind(this)
    );
  }
  componentDidMount() {
    const { searchTerm } = this.state;
    this.searchByRequest(searchTerm);
  }

  render() {
    const { searchTerm } = this.state;
    return (
      <div className="page-content">
        <Search
          value={searchTerm}
          onChange={this.onSearchChange}
          onSubmit={this.onSearchSubmit}
        />
        <div>
          {!(this.state.title && this.state.url && this.state.description)
            ? <p className="preloadMessage">Type smth & hit enter</p>
            : <ArticlesGrid
                title={this.state.title}
                description={this.state.description}
                url={this.state.url}
              />}
        </div>
      </div>
    );
  }
}
export default App;

//
