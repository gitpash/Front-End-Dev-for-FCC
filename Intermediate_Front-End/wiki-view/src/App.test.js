import React from "react";
import App from "./App";
import renderer from "react-test-renderer";

test("renders without crashing", () => {
  const tree = renderer.create(<App />);
  expect(tree).toMatchSnapshot();
});
