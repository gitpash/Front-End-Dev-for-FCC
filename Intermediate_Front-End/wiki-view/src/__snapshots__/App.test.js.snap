exports[`test renders without crashing 1`] = `
<div
  className="page-content">
  <form
    onSubmit={[Function]}>
    <input
      autoFocus={true}
      onChange={[Function]}
      type="text"
      value="" />
    <button
      type="submit">
      go!
    </button>
  </form>
  <div>
    <p
      className="preloadMessage">
      Type smth & hit enter
    </p>
  </div>
</div>
`;
